.github/workflows/end-to-end-tests.yaml

```diff
@@ -133,6 +133,11 @@ jobs:
             deployments: write
         steps:
             - uses: actions/checkout@v4
+              with:
+                  ref: ${{ steps.sha.outputs.sha }}
```

Comment:
You can't reference steps in other jobs, you'd have to make it an output in that job if that's what you wanted to achieve

Reply:
I jst copied it from above so probably cargoculting - I guess this and the other one are just unnecessary?

Reply:
the other one is necessary, if your PR is not based off latest you don't want to run the tests for the default branch, you want to run the tests you wrote in your PR.